src/main/java/frc/robot/AutoFactory.java

```diff
@@ -367,6 +370,20 @@ public Command getWingAndMidlineAuto() {
 
     }
 
+    public Command autoPickupNote() {
+        Pose2d[] notePositions = driveBase.getNotePositions();
+        Pose2d pose = driveBase.getPose();
+        var selectedNoteOptional = Stream.of(notePositions)
+                .sorted(Comparator.comparingDouble(notePose -> notePose.minus(pose).getTranslation().getNorm()))
+                .findFirst();
+        if (selectedNoteOptional.isEmpty()) return new InstantCommand();
+
+        var selectedNote = selectedNoteOptional.get();
+
+        return new TurnToPointCommand(driveBase, driveBase::getPose, selectedNote, 0, 0, true, true)
+                .andThen(new IntakeNoteCommand(indexer, intake).deadlineWith(getPathFindToPoseCommand(selectedNote)));
```

Comment:
I'd be careful about ending it exactly when you drive to the note? Might be a good idea to drive a bit "over" the note and keep intaking so you know you picked it up